x-pack/filebeat/input/awss3/input.go

```diff
@@ -410,5 +413,31 @@ func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
 	}
 }
 
+func pollSqsUtilizationMetric(ctx context.Context, receiver *sqsReader, metricReporterChan chan int64) {
+	pollDur := 5 * time.Second
+	t := time.NewTicker(pollDur)
+	defer t.Stop()
+
+	utilizedNanos := new(int64)
+	go func() {
+		for elem := range metricReporterChan {
+			*utilizedNanos += elem
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			denom := pollDur.Nanoseconds() * int64(receiver.maxMessagesInflight)
```

Comment:
I think this is overly optimistic for the accuracy of the ticker. If you keep a `lastTick time.Time` in the encompassing scope, you can calculate the time delta with

```
lastTick := time.Now()
for {
    ...
    case tick := <-t.C:
        denom := int64(tick.Sub(lastTick)) * int64(receiver.maxMessagesInflight)
        lastTick = tick
```
Maybe also do the float conversion here rather than in the next line (`s/int64/float64/g`)